flair/embeddings.py

```diff
@@ -354,23 +354,23 @@ def __init__(self, model: str = 'original'):
 
         self.name = 'elmo-' + model
         self.static_embeddings = True
-
-        # the default model for ELMo is the 'original' model, which is very large
-        options_file = allennlp.commands.elmo.DEFAULT_OPTIONS_FILE
-        weight_file = allennlp.commands.elmo.DEFAULT_WEIGHT_FILE
-        # alternatively, a small, medium or portuguese model can be selected by passing the appropriate mode name
-        if model == 'small':
-            options_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x1024_128_2048cnn_1xhighway/elmo_2x1024_128_2048cnn_1xhighway_options.json'
-            weight_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x1024_128_2048cnn_1xhighway/elmo_2x1024_128_2048cnn_1xhighway_weights.hdf5'
-        if model == 'medium':
-            options_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x2048_256_2048cnn_1xhighway/elmo_2x2048_256_2048cnn_1xhighway_options.json'
-            weight_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x2048_256_2048cnn_1xhighway/elmo_2x2048_256_2048cnn_1xhighway_weights.hdf5'
-        if model == 'pt' or model == 'portuguese':
-            options_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/contributed/pt/elmo_pt_options.json'
-            weight_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/contributed/pt/elmo_pt_weights.hdf5'
-        if model == 'pubmed':
-            options_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/contributed/pubmed/elmo_2x4096_512_2048cnn_2xhighway_options.json'
-            weight_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/contributed/pubmed/elmo_2x4096_512_2048cnn_2xhighway_weights_PubMed_only.hdf5'
+        if not options_file or not weight_file:
```

Comment:
I think we can just initialize both `options_file` and `weight_file` via constructor argument:

```python
def __init__(self, model: str = 'original', 
                    options_file: str = allennlp.commands.elmo.DEFAULT_OPTIONS_FILE,
                    weight_file: str = allennlp.commands.elmo.DEFAULT_WEIGHT_FILE):
```

Then it is semantically clear, that these default values are used + no `if not` checks are needed here :)

Reply:
@stefan-it 
Hi, 
thanks for the review.

This is what i initially wanted to do, but the code currently imports allennlp only at a later stage, in a try except block
```python
try:
      import allennlp.commands.elmo
except:
```
so the default options and weight files are unavailable in the constructor.
Any suggestions?

Reply:
Oh, sorry. I forgot that we import `allennlp` later...